File: typescript/pascals-triangle/pascals-triangle.ts
export default class Triangle {
  private size: number;

  constructor(size: number) {
    this.size = size;
  }

  get rows(): number[][] {
    let acc = [];

    for (let i = 1; i <= this.size; i++) acc.push(this.row(i));

    return acc;
  }

  get lastRow(): number[] {
    return this.row(this.size);
  }

  row(n: number): number[] {
    let digits = [1];
    let column = 1;

    for (let j = 1; j < n; j++) {
      column = (column * (n - j)) / j;
      digits.push(column);
    }

    return digits;
  }
}

File: kotlin/pascals-triangle/src/main/kotlin/PascalsTriangle.kt
object PascalsTriangle {
    fun computeTriangle(rows: Int): List<List<Int>> {
        require(rows >= 0) { "Rows can't be negative!" }
        return 1.rangeTo(rows).map(::row)
    }

    private fun row(i: Int) =
        sequence {
            var column = 1
            yield(column)

            for (j in 1 until i) {
                column = column * (i - j) / j
                yield(column)
            }
        }.toList()
}